### PR TITLE
Make use of bech32 encoding when displaying addresses

### PR DESCRIPTION
**Pull request by fixone**
First place to make use of it - keygenerator

### CODE REVIEW

File: data/state/addressConverters/plainAddressConverter.go [hunk 1]
```diff
@@ -106,3 +108,45 @@ func (pac *PlainAddressConverter) PrepareAddressBytes(addressBytes []byte) ([]by
...
+	}
+
+	_, dec, err := bech32.Decode(bech32Address)
+	if err != nil {
+		//TODO move error to state/errors.go
+		return nil, errors.New("wrong bech32 string: " + bech32Address)
+	}
+	conv, err := bech32.ConvertBits(dec, 5, 8, false)
+	if err != nil {
+		//TODO move error to state/errors.go
```
Comment (sasurobert) [inline data/state/addressConverters/plainAddressConverter.go]: resolve TODO's :)

File: data/state/addressConverters/plainAddressConverter.go [hunk 2]
```diff
@@ -106,3 +108,45 @@ func (pac *PlainAddressConverter) PrepareAddressBytes(addressBytes []byte) ([]by
...
+	if len(bech32Address) == 0 {
+		return nil, state.ErrEmptyAddress
+	}
+
+	_, dec, err := bech32.Decode(bech32Address)
+	if err != nil {
+		//TODO move error to state/errors.go
+		return nil, errors.New("wrong bech32 string: " + bech32Address)
+	}
+	conv, err := bech32.ConvertBits(dec, 5, 8, false)
```
Comment (sasurobert) [inline data/state/addressConverters/plainAddressConverter.go]: add an empty line before.

File: data/state/addressConverters/plainAddressConverter.go [hunk 3]
```diff
@@ -106,3 +108,45 @@ func (pac *PlainAddressConverter) PrepareAddressBytes(addressBytes []byte) ([]by
...
+const erd = "erd"
+
+// ConvertToBech32 returns the address in bech32 format
+func (pac *PlainAddressConverter) ConvertToBech32(addressContainer state.AddressContainer) (string, error) {
+	if addressContainer == nil {
+		return "", state.ErrNilAddressContainer
+	}
+
+	conv, err := bech32.ConvertBits(addressContainer.Bytes(), 8, 5, true)
+	if err == nil {
```
Comment (sasurobert) [inline data/state/addressConverters/plainAddressConverter.go]: change to verify if err != nil then return error. like in createaddressFromBech32

File: cmd/keygenerator/main.go [hunk 4]
```diff
@@ -127,8 +128,18 @@ func generateFiles(ctx *cli.Context) error {
 	}
 
 	fmt.Println("Files generated successfully.")
-	fmt.Printf("   pk for balance: %s\n", pkHexBalance)
-	fmt.Printf("   pk for block signing: %s\n", pkHexBlockSigning)
+	fmt.Printf("\tpk for balance:\t%s\n", pkHexBalance)
+	ac, _ := addressConverters.NewPlainAddressConverter(32, "")
```
Comment (sasurobert) [inline cmd/keygenerator/main.go]: treat the returned errors.

File: cmd/keygenerator/main.go [hunk 5]
```diff
@@ -127,8 +128,18 @@ func generateFiles(ctx *cli.Context) error {
...
+	adr, _ := ac.CreateAddressFromHex(pkHexBalance)
+	bech32, _ := ac.ConvertToBech32(adr)
+	fmt.Printf("\tpk in bech32:\t%s\n", bech32)
+	fmt.Printf("\tpk for block signing:\t%s\n", pkHexBlockSigning)
+	//the block signing PK results in a bech32 string greater than the standard imposed 90 char limit
+	//but signing key is longer and can't be mistaken for a txid as it is the case with the balance one
+	//ac, _ = addressConverters.NewPlainAddressConverter(128, "")
```
Comment (sasurobert) [inline cmd/keygenerator/main.go]: delete unused code.

File: data/state/addressConverters/plainAddressConverter.go (hunk 2 above)
Comment (fixone) [reply data/state/addressConverters/plainAddressConverter.go]: done

File: data/state/addressConverters/plainAddressConverter.go (hunk 3 above)
Comment (fixone) [reply data/state/addressConverters/plainAddressConverter.go]: done

File: cmd/keygenerator/main.go (hunk 4 above)
Comment (fixone) [reply cmd/keygenerator/main.go]: done

File: cmd/keygenerator/main.go (hunk 5 above)
Comment (fixone) [reply cmd/keygenerator/main.go]: deleted code, left comment to see why we did not use the encoding for longer keys

File: cmd/keygenerator/main.go [hunk 6]
```diff
@@ -127,8 +128,19 @@ func generateFiles(ctx *cli.Context) error {
 	}
 
 	fmt.Println("Files generated successfully.")
-	fmt.Printf("   pk for balance: %s\n", pkHexBalance)
-	fmt.Printf("   pk for block signing: %s\n", pkHexBlockSigning)
+	fmt.Printf("\tpk for balance:\t%s\n", pkHexBalance)
+	ac, _ := addressConverters.NewPlainAddressConverter(32, "")
```
Comment (sasurobert) [inline cmd/keygenerator/main.go]: treat errors.

File: cmd/keygenerator/main.go [hunk 7]
```diff
@@ -127,8 +128,19 @@ func generateFiles(ctx *cli.Context) error {
...
+	adr, _ := ac.CreateAddressFromHex(pkHexBalance)
```
Comment (sasurobert) [inline cmd/keygenerator/main.go]: treat error

**Pull request merged**